backend/kesaseteli/applications/api/v1/views.py

```diff
@@ -1,16 +1,39 @@
 from rest_framework import status
+from rest_framework.exceptions import PermissionDenied
 from rest_framework.response import Response
 from shared.audit_log.viewsets import AuditLoggingModelViewSet
 
 from applications.api.v1.serializers import ApplicationSerializer
+from applications.enums import ApplicationStatus
 from applications.models import Application
 
 
 class ApplicationViewSet(AuditLoggingModelViewSet):
-    queryset = Application.objects.select_related("company").prefetch_related(
-        "summer_vouchers"
-    )
+    queryset = Application.objects.all()
     serializer_class = ApplicationSerializer
 
+    def get_queryset(self):
+        """
+        Fetch all DRAFT status applications of the company.
+        Should inlcude only 1 application since we don't allow creation of multiple
+        DRAFT applications per company.
+        """
+        queryset = (
+            super()
+            .get_queryset()
+            .select_related("company")
+            .prefetch_related("summer_vouchers")
+        )
+        user_company = self.request.user.oidc_profile.eauthorization_profile.company
+        return queryset.filter(company=user_company, status=ApplicationStatus.DRAFT)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Allow only 1 (DRAFT) application per company.
+        """
+        if self.get_queryset().exists():
+            raise PermissionDenied("Company can have only one draft application")
```

Comment:
perhaps nitpick but permission denied usually means that user has no right to do an operation (403). But this is just illegal operation (400 or 422 or something).

Reply:
Good point! I'll change this to `ValidationError` (400).